### DISCUSSION TITLE
BigQuad Deck - Using Battery monitoring with 6S LiPo

### DISCUSSION THREAD

**Discussion by schlagenhauf** [bitcraze/discussions#98] [Q&A]
Hi,

my BigQuad-based quadcopter uses a 6S LiPo battery. I'd like to connect the battery voltage monitor, but the BigQuad product page says the maximum voltage is 23V which a 6S LiPo exceeds when fully charged (4.2V * 6 = 25.2V).

Looking at the BigQuad schematics, the voltage divider would bring the voltage down to 3.2V on `VBAT_DIV`, which is higher than the 3V of the system, but below 3.3V which [seems to be supported](https://www.bitcraze.io/documentation/hardware/crazyflie_2_1/crazyflie_2_1-datasheet.pdf)(p.2, Section 7).

I don't really care about an accurate `VBAT` measurement when the battery is in this range, but i'd like to make sure i don't fry the microcontroller. Could you tell me if i'm on the right track or if i rather should not do that?

Thanks!

**Comment by tobbeanton** [comment]
If I understand the STM32F405 datasheet correctly the input is tolerant to at least 4.0V and more if it is a 5V tolerant pin so you should be safe there. You will just get a saturated value for anything above Vref which is 3V.

**Reply by schlagenhauf** [reply]
Thanks for your answer! It "works" in so far that nothing bad happend so that's great. I get weird voltage readings in cfclient, switching between ~4V and ~0.1 V. I'll be back with more details after some more debugging.

**Reply by tobbeanton** [reply]
That is a bit weird, possibly that might be what happens if ADC is saturated or maybe a overflow.
